### ISSUE TITLE
Option 'Exclude page from SearchEngines' is not checked in SitemapIndexHook

### ISSUE THREAD

**Issue by thomaszbz** [mblaschke/TYPO3-metaseo#127]
by Georg Tiefenbrunn via https://forge.typo3.org/issues/61858

If the option 'Exclude page from SearchEngines' is checked in page properties the page is excluded in the sitemaps (which is the desired/expected behavior).

SitemapIndexHook does not respect this setting and always populates table tx_metaseo_sitemap. This leads to an inconsistency between the generated sitemaps (pages/URIs are excluded) and the listing in module "Sitemap" (pages/URIs are included).

Desired/expected behavior: SitemapIndexHook should only populate table tx_metaseo_sitemap if option 'Exclude page from SearchEngines' is not set.

**Comment by thomaszbz** [comment]
Updated by Markus Blaschke 11 months ago

    Category set to Sitemap
    Status changed from New to Needs Feedback
    Assigned To set to Markus Blaschke
    Target version set to 2.0.0

Comment

You're right, that field is not checked while indexing the page.
But the check is done in sitemap.xml output.

I've added an excluded column in the sitemap backend module so it's visible there if an sitemap entry is excluded by this option (Commit develop e4ac66e)
#2
Updated by Georg Tiefenbrunn 10 months ago
Comment

Servus Markus!

    I've added an excluded column in the sitemap backend module so it's visible there if an sitemap entry is excluded by this option

That's nice. But I still think it would be better to not populate table tx_metaseo_sitemap with unwanted data.

I use a preview page for EXT:news articles which has 'Exclude page from SearchEngines' set. Only BE users will ever see this page while creating news articles. At the moment every news article is listed twice in the SEO > Sitemaps module: the unwanted and unnecessary preview link (which is awfully long like preview-page/?tx_news_pi1%5Bnews%5D=1&tx_news_pi1%5Bday%5D=25&tx_news_pi1%5Bmonth%5D=08&tx_news_pi1%5Byear%5D=2014&cHash=00000000000000000000000000000000) and the correctly index page/article.

Imagine a site with other extensions like media gallery, products, etc. The sitemap module would be filled with lots of unwanted data hiding the important data.
#3
Updated by Markus Blaschke 10 months ago

    Status changed from Needs Feedback to Accepted
    Priority changed from Should have to Must have

#4
Updated by Markus Blaschke 10 months ago

    Status changed from Accepted to Needs Feedback
    % Done changed from 0 to 50

Comment

In theory the solution should be easy but MetaSEO also hooks into the typolink method so each link is indexed.
Because of this feature MetaSEO have to fetch the exclude-information for each of this pages but i want to void the overhead of the additional queries in the typolink hook.

But I've added a check in the current page (only checks the current page information) in commit a703006.
This check prevents files to be indexed which are used on this page (file indexer will be available in MetaSEO 2.0).

I think the current solution is the fastest one.
#5
Updated by Markus Blaschke 10 months ago

    Complexity set to medium

**Comment by thomaszbz** [comment]
@georg-tiefenbrunn is this still an issue?

**Comment by thomaszbz** [comment]
It's repruced, fixed, merged and nobody complains. Closing this issue now.

**Comment by thomaszbz** [comment]
As it turns out, a703006 did _not_ get merged into metaseo yet (as of unreleased 2.0.2). Reopening this issue.

The corresponding branch is still there, but not clean: https://github.com/mblaschke/TYPO3-metaseo/tree/feature/forge-61858

**Comment by thomaszbz** [comment]
Merge conflicts for a703006 solved via PR #304 / ddead84.

@mblaschke could you please review/merge your commit via #304? I did not test it so far.

**Comment by thomaszbz** [comment]
Alternatively: @georg-tiefenbrunn , have you been testing the patch? I know, it's long time ago...

**Comment by georg-tiefenbrunn** [comment]
I can test it on 27.12.

**Comment by thomaszbz** [comment]
great! thanks!

**Comment by thomaszbz** [comment]
@georg-tiefenbrunn Any progress so far?

**Comment by georg-tiefenbrunn** [comment]
@thomaszbz Sorry, this took a while longer than expected.

tl;dr: Works as it should with your patch applied. Thanks for bringing up and fixing this old one!

I got confused by the "Excl." column in the sitemap backend module introduced with e4ac66e and double checked the indexing.

My first thought was that the "Excl." column is now obsolete, but after playing around with enabling/disabling the option 'Exclude page from SearchEngines' found it very useful. If a page got indexed without the exclude option set an entry is created in table `tx_metaseo_sitemap`. If you later decide to exclude the page it's removed from the sitemap output (:ok_hand:, as it did before) and you can easily remove it using the sitemap backend module (:thumbsup:).

Using a preview page for EXT:news articles now don't add garbage to table `tx_metaseo_sitemap`.

Again, thanks for fixing this oldie!
